.github/workflows/multi-platform-test.yml

```diff
@@ -0,0 +1,328 @@
+      if: github.event.inputs.submodules_json  != ''
+      working-directory: .ci_cd
+      run: ./checkout_submodules.sh '${{ github.event.inputs.submodules_json }}'
+
+    - name: Determine submodule path
+      id: get-submodule-path
+      run: |
+        # Parsing .gitmodules to find the path of the submodule repository.
+        submodule_path=$(git config --file .gitmodules --get-regexp path | awk -v repo_name=${{ github.event.repository.name }} '$0 ~ repo_name {print $2}')
+        echo "submodule-path=$submodule_path" >> $GITHUB_OUTPUT
+
+    - name: Checkout current repository as umbrella submodule lib
+      uses: actions/checkout@v4
+      with:
+        path: ${{ steps.get-submodule-path.outputs.submodule-path }}
+        fetch-depth: 1
+
+    - name: Set usefull strings
+      # Turn input strings (such as the build output directory) into step outputs. These step outputs can be used throughout the workflow file.
+      id: strings
+      shell: bash
+      run: |
+        # CMake locations
+        umbrella_build_dir="${{ github.workspace }}/build"
+        submodule_build_dir="$umbrella_build_dir/${{ steps.get-submodule-path.outputs.submodule-path }}"
+        echo "umbrella-build-dir=$umbrella_build_dir" >> "$GITHUB_OUTPUT"
+        echo "submodule-build-dir=$submodule_build_dir" >> "$GITHUB_OUTPUT"
+        echo "submodule-test-dir=$submodule_build_dir/test" >> "$GITHUB_OUTPUT"
+
+        # Dependencies dir location
+        echo "dependencies-dir=${{ github.workspace }}/../dependencies" >> "$GITHUB_OUTPUT"
+
+        # Get platform version
+        os_version=${{ matrix.os }}
+        platform_version=${os_version##*-}
+        echo "platform-version=${platform_version}" >> $GITHUB_OUTPUT
+
+    - name: Set up dependencies directory
+      run: |
+        mkdir -p "${{ steps.strings.outputs.dependencies-dir }}"
+
+    - name: Install boost
+      uses: MarkusJx/install-boost@v2.4.4
+      id: install-boost
+      with:
+          boost_version: ${{ env.BOOST_VERSION }}
+          # Extract version number from matrix.os by replacing os_name-
+          # A list of supported versions can be found here:
+          # https://github.com/MarkusJx/prebuilt-boost/blob/main/versions-manifest.json
+          platform_version: ${{ steps.strings.outputs.platform-version }}
+          boost_install_dir: ${{ steps.strings.outputs.dependencies-dir }}
+
+    - name: Configure CMake
+      run: >
+        cmake -B ${{ steps.strings.outputs.umbrella-build-dir }}
+        -DCMAKE_CXX_COMPILER=${{ matrix.cpp_compiler }}
+        -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
+        -DBUILD_TESTS=TRUE
+        -DENABLE_JUNIT_TEST_OUTPUT=TRUE
+        -S ${{ github.workspace }}
+      env:
+        BOOST_ROOT: "${{ steps.install-boost.outputs.BOOST_ROOT }}"
+
+    - name: Build all inside submodule
+      working-directory: ${{ steps.strings.outputs.submodule-build-dir }}
+      run: |
+        #TODO: use cmake build instead?
+        make all
```

Comment:
Are we sure make all will work? Half our targets don't build usually, and we don't remove them yet. Another small part builds but does not pass the tests, and we are ok with it...

Reply:
Made it not failing in case of build errors. Build log is parsed and converted into JUnit result. Then handled as a separate test